prow/flagutil/kubernetes_cluster_clients.go

```diff
@@ -346,7 +350,7 @@ var clientCreationFailures = prometheus.NewCounterVec(prometheus.CounterOpts{
 // BuildClusterManagers returns a manager per buildCluster.
 // Per default, LeaderElection and the metrics listener are disabled, as we assume
 // that there is another manager for ProwJobs that handles that.
-func (o *KubernetesOptions) BuildClusterManagers(dryRun bool, callBack func(), opts ...func(*manager.Options)) (map[string]manager.Manager, error) {
+func (o *KubernetesOptions) BuildClusterManagers(dryRun bool, requiredTestPodVerbs []string, callBack func(), opts ...func(*manager.Options)) (map[string]manager.Manager, error) {
```

Comment:
Just a side note for future work in this area:
At some point it might be worth moving the build cluster status tracking logic here rather than determining build cluster status separately. Now that we're checking the authorization here we are already checking for 'reachability' in a more robust way and could return more precise information about what permission is missing for a cluster.
Here is the existing status tracking logic I'm referring to: https://github.com/kubernetes/test-infra/blob/e39fd1ac2edfaf14635705374030734a25415828/prow/plank/reconciler.go#L202-L255

Reply:
Ack.

Reply:
@cjwagner I think I've done this locally (at least a first stab at this). Please have a look at https://github.com/kubernetes/test-infra/pull/29236 to get the ball rolling.

FTR I am trying out a new PR authoring style where I try to create just 1 commit per PR, pretty much like how Gerrit does it because then it (1) results in smaller PRs and (2) allows me to make faster iterative edits because I'm not amending a commit buried somewhere older than HEAD. I got 2 more commits locally but will submit them for review only after we've merged the one above.